src/Waives.Pipelines/Pipeline.cs

```diff
@@ -226,26 +211,68 @@ public Pipeline OnDocumentError(Action<DocumentError> action)
         /// this object.</returns>
         public IDisposable Start()
         {
-            _pipeline = _pipeline.SelectMany(async d =>
+            var (disposable, _) = StartAsync();
+            return disposable;
+        }
+
+        private (IDisposable, Task) StartAsync()
+        {
+            // take a copy of _docActions so Start() is idempotent
+            var docActions = new List<Func<WaivesDocument, Task<WaivesDocument>>>(_docActions)
             {
-                await d.HttpDocument.Delete(() =>
+                async d =>
                 {
-                    _rateLimiter.MakeDocumentSlotAvailable();
-                    _logger.Log(LogLevel.Info, $"Completed processing '{d.Source.SourceId}' and deleted Waives document");
-                }).ConfigureAwait(false);
+                    await d.HttpDocument.Delete().ConfigureAwait(false);
```

Comment:
💪 I'm not keen on this essential "`finally`" action being part of the collection of actions run; I'd much prefer it was kept separate to ensure it isn't ever accidentally removed from the collection, and can be explicitly invoked to maximise clarity.